src/cloud/components/Views/Table/TableViewContentManager.tsx

```diff
@@ -252,6 +254,67 @@ const TableViewContentManager = ({
     [setPreferences]
   )
 
+  const [newFolderRowState, setNewFolderRowState] = useState<
+    'idle' | 'editing' | 'submitting'
+  >('idle')
+  const [newFolderName, setNewFolderName] = useState('')
+  const compositionStateRef = useRef(false)
+  const newFolderInputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    if (newFolderInputRef.current != null && newFolderRowState === 'editing') {
+      newFolderInputRef.current.focus()
+    }
+  }, [newFolderRowState])
+
+  const createNewFolder = useCallback(async () => {
+    setNewFolderRowState('submitting')
+    if (currentWorkspaceId != null) {
+      await createFolder(
+        team,
+        {
+          folderName: newFolderName,
+          description: '',
+          workspaceId: currentWorkspaceId,
+          parentFolderId: currentFolderId,
+        },
+        { skipRedirect: true }
+      )
+    }
+    setNewFolderName('')
+    setNewFolderRowState('idle')
+  }, [currentWorkspaceId, currentFolderId, createFolder, team, newFolderName])
+
+  const [newDocRowState, setNewDocRowState] = useState<
+    'idle' | 'editing' | 'submitting'
+  >('idle')
+  const [newDocName, setNewDocName] = useState('')
+  const newDocCompositionStateRef = useRef(false)
+  const newDocInputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    if (newDocInputRef.current != null && newDocRowState === 'editing') {
+      newDocInputRef.current.focus()
+    }
+  }, [newDocRowState])
+
+  const createNewDoc = useCallback(async () => {
+    setNewDocRowState('submitting')
+    if (currentWorkspaceId != null) {
```

Comment:
these buttons already show up with 
```
{currentWorkspaceId != null && (
            <TableContentManagerRow>
```
so imo we are fine either assert it or pass it down via function parameter

Reply:
I did it intentionally to prevent further problems since the code is too big and the function is exposed regardless the existence of `currentWorkspaceId` prop.

But I think we can deal with this issue more elegantly by extracting the whole new doc/folder components. I'll do along with https://github.com/BoostIO/BoostNote-App/pull/1351#discussion_r755526951